### DISCUSSION TITLE
SPDIF clock rates

### DISCUSSION THREAD

**Discussion by dhedgeland** [pschatzmann/arduino-audio-tools#655] [Q&A] [answered]
I'm trying to read MP3 files from an SD card and output SPDIF using the Audio Tools on an ESP32.  The MP3 files are encoded at 44.1 kHz and I get output but the output rate seems to be about 10% slow. Output using I2S with an external DAC seems good.
I've tested with the streams generator examples and confirmed that with I2S output they seem to work well with sample rates of both 44.1 and 48 kHz. With SPDIF output 48 kHz seems good but at 44.1 kHz the actual output rate seems to be slow as mentioned before.
I'll investigate further but I thought I would ask now in case anybody has seen the same effect.
BTW I'm very impressed by the SPDIF generation in AudioSPDIF.h - very neat !

**Comment by pschatzmann** [comment, marked as answer]
That's strange: if I got you right, it is working with the sine generator, but not via mp3 decoding on 44100.

You could try to play around by adjusting the sample_rate a bit on line 154. (Or you could try to generate more samples with the help of a ResampleStream)

Unfortunatly, I can't test this myself because I do not have any SPDIF equipment.
